test/python/opflow/test_abelian_grouper.py

```diff
@@ -45,6 +47,42 @@ def test_abelian_grouper(self, pauli_op):
             num_groups = 4
         grouped_sum = AbelianGrouper().convert(paulis)
         self.assertEqual(len(grouped_sum.oplist), num_groups)
+        for group in grouped_sum:
+            for op_1, op_2 in combinations(group, 2):
+                self.assertTrue(
+                    (op_1 @ op_2 - op_1 @ op_2).reduce().primitive.coeffs[0] == 0
+                )
+                # uncomment after #5537
+                # self.assertTrue(commutator(op_1, op_2).is_zero())
+
+    @data("h2_op", "generic")
+    def test_abelian_grouper_summedop(self, pauli_op):
```

Comment:
It's good to add an option `pauli_sum_op` to test both `SummedOp` and `PauliSumOp`.